### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by reislerjul**
Made changes to allow the FLAG and FLAGCallback classes to appear in the docs.

To generate and view the docs:
1. Install Sphinx: `pip install git+https://github.com/pyg-team/pyg_sphinx_theme.git`
2. cd into the docs folder: `cd docs`
3. Make the docs: `make html`
4. Open `docs/build/html/index.html` in a browser. The FLAG docs are in the `torch_geometric.contrib` section.

### CODE REVIEW

**Comment by reislerjul** [comment]
Screen Shot 2023-03-11 at 11 42 32 AM
Screen Shot 2023-03-11 at 11 42 23 AM
Screen Shot 2023-03-11 at 11 42 13 AM

File: torch_geometric/contrib/flag/flag.py [hunk 1]
```diff
@@ -42,41 +102,18 @@ def forward(
         # Code below adapted from:
         #   https://github.com/devnkong/FLAG/blob/main/deep_gcns_torch
         #     /examples/ogb/ogbn_arxiv/main.py#L56
+        #   and
+        #   https://github.com/devnkong/FLAG/blob/main/deep_gcns_torch
+        #     /examples/ogb/attacks.py
         #
         # This example does not assume that the model forward function has a
         #   `perturb` argument
-        # forward = lambda perturb : model(x+perturb, edge_index)[train_idx]
```
Comment (reislerjul) [inline torch_geometric/contrib/flag/flag.py:49]: @tnichols89 I removed a lot of these comments; let me know if you want to keep them in.

Comment (tnichols89) [reply torch_geometric/contrib/flag/flag.py:49]: Certainly do not want to keep them in! Thanks for removing them.

**Comment by tnichols89** [comment]
Is this branch intentionally being merged into Jamil's branch?

I'll be taking a look at the docs themselves here shortly.

File: torch_geometric/contrib/flag/flag.py [hunk 2]
```diff
@@ -1,17 +1,51 @@
...
+    end of the steps, the model parameters are updated with these accumulated
+    gradients.
+
+    FLAG can be used in node classification, link prediction, and graph
+    classification. However, this implementation is only meant for data with
+    continuous node features (while the original paper handles discrete node
+    features, it does so through changes to the model's `forward` function).
+
+    .. note::
+        For examples of using FLAG in a training loop, see
```
Comment (tnichols89) [inline torch_geometric/contrib/flag/flag.py]: @reislerjul Just want to make sure we actually have an example available here by the time this gets merged if we're going to leave this note in the docs.

File: torch_geometric/contrib/flag/flag.py [hunk 3]
```diff
@@ -24,13 +58,39 @@ def __init__(
...
+        edge_index: torch.Tensor,
+        y_true: torch.Tensor,
+        train_idx: torch.Tensor,
         step_size: float,
         n_ascent_steps: int = 3,
-    ) -> tuple[torch.nn.Module, torch.tensor]:
+    ) -> tuple[torch.nn.Module, torch.Tensor]:
+        """
+        Perform the FLAG forward and backward passes. Note that users should
+        not invoke loss.backward() or optimizer.step() externally to this
```
Comment (tnichols89) [inline torch_geometric/contrib/flag/flag.py]: Are we able to wrap `loss.backward()` and `optimizer.step()` in backticks to get inline code formatting in Sphinx?

**Comment by tnichols89** [comment]
Other than the couple minor comments above, the docs look good to me.

File: torch_geometric/contrib/flag/flag.py (hunk 2 above)
Comment (reislerjul) [reply torch_geometric/contrib/flag/flag.py]: Good point! Let me remove this for now.

File: torch_geometric/contrib/flag/flag.py (hunk 3 above)
Comment (reislerjul) [reply torch_geometric/contrib/flag/flag.py]: Should be able to! Will add
